pkg/operator/ceph/object/realm/controller.go

```diff
@@ -176,6 +198,31 @@ func (r *ReconcileObjectRealm) reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileObjectRealm) pullCephRealm(realm *cephv1.CephObjectRealm) (reconcile.Result, error) {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", realm.Name)
+	urlArg := fmt.Sprintf("--url=%s", realm.Spec.Pull.Endpoint)
+	logger.Debug("getting keys to pull realm")
+	accessKeyArg, secretKeyArg, err := object.GetRealmKeyArgs(r.context, realm.Name, realm.Namespace)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return waitForRequeueIfRealmKeysNotReady, err
+		}
+		return waitForRequeueIfRealmKeysNotReady, errors.Wrap(err, "failed to get keys for realm")
+	}
+	logger.Debugf("keys found to pull realm, getting ready to pull from endpoint %q", realm.Spec.Pull.Endpoint)
+
+	objContext := object.NewContext(r.context, realm.Name, realm.Namespace)
+
+	output, err := object.RunAdminCommandNoRealm(objContext, "realm", "pull", realmArg, urlArg, accessKeyArg, secretKeyArg)
```

Comment:
@travisn Do you think I should do a `realm get` to make sure the realm doesn't already exist on this cluster before I do the realm pull?

Reply:
What is the side effect pulling if it already exists? Is it a no-op? If it's more efficient to do a `get` first, that sounds better.